pkg/cmd/issue/view/view.go

```diff
@@ -86,18 +89,39 @@ func viewRun(opts *ViewOptions) error {
 		return err
 	}
 
+	var baseRepo ghrepo.Interface
+	if _, r := shared.IssueMetadataFromURL(opts.SelectorArg); r != nil {
+		baseRepo = r
+	} else {
+		baseRepo, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+	}
+
 	lookupFields := set.NewStringSet()
 	if opts.Exporter != nil {
 		lookupFields.AddValues(opts.Exporter.Fields())
 	} else if opts.WebMode {
 		lookupFields.Add("url")
 	} else {
+		if opts.Comments {
+			lookupFields.Add("comments")
+			lookupFields.Remove("lastComment")
+		}
+		if opts.Detector == nil {
+			cachedClient := api.NewCachedHTTPClient(httpClient, time.Hour*24)
+			opts.Detector = fd.NewDetector(cachedClient, baseRepo.RepoHost())
+		}
+		features, err := opts.Detector.IssueFeatures()
+		if err != nil {
+			return err
+		}
+		if features.StateReason {
+			lookupFields.Add("stateReason")
```

Comment:
Do you think it would it be simpler to push this logic into `shared.findIssueOrPR`? 

Example: if `findIssueOrPR` was invoked with `stateReason` among fields, use featuredetector to determine whether it's supported, and scrub it out if not. Prior art: https://github.com/cli/cli/blob/e2e8d697dbe97e675dac173bfddea523beafc7d7/pkg/cmd/pr/shared/finder.go#L142-L151

This way, all callers of `findIssueOrPR` across all commands would not have to worry about GHES not supporting certain fields.

The only gotcha would be that a `fd.Detector` would have to be passed down as an injectable, though 🤔 Feel free to disregard if that's too much hassle.

Reply:
I like this idea. I went ahead and implemented it. Luckily (or unluckily) none of our issue tests target GHES so the feature detection here is a no-op and doesn't need to be injected for testing purposes.

Reply:
Future tests that target GHES could also just mock out the request instead of having to inject the detector.